src/components/application_manager/test/commands/command_request_impl_test.cc

```diff
@@ -0,0 +1,485 @@
+  (*msg)[strings::msg_params] = 0u;
+
+  CommandPtr command = CreateCommand<UCommandRequestImpl>(msg);
+
+  MockAppPtr app = CreateMockApp();
+  ApplicationSet app_set;
+  app_set.insert(app);
+
+  DataAccessor<ApplicationSet> accessor(app_set, app_set_lock_);
+
+  EXPECT_CALL(app_mngr_, applications()).WillOnce(Return(accessor));
+  EXPECT_CALL(*app, app_id()).WillOnce(Return(kConnectionKey_));
+  EXPECT_CALL(*app, policy_app_id()).WillOnce(Return(kPolicyAppId_));
+  EXPECT_CALL(*app, hmi_level())
+      .WillOnce(Return(mobile_apis::HMILevel::HMI_NONE));
+
+  EXPECT_CALL(app_mngr_, CheckPolicyPermissions(_, _, _, _, _))
+      .WillOnce(Return(kMobResultSuccess_));
+
+  EXPECT_TRUE(command->CheckPermissions());
+}
+
+TEST_F(CommandRequestImplTest,
+       CheckAllowedParameters_WrongPolicyPermissions_UNSUCCESS) {
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::connection_key] = kConnectionKey_;
+  (*msg)[strings::msg_params] = 0u;
+
+  CommandPtr command = CreateCommand<UCommandRequestImpl>(msg);
+
+  MockAppPtr app = CreateMockApp();
+  ApplicationSet app_set;
+  app_set.insert(app);
+
+  DataAccessor<ApplicationSet> accessor(app_set, app_set_lock_);
+
+  EXPECT_CALL(app_mngr_, applications()).WillOnce(Return(accessor));
+  EXPECT_CALL(*app, app_id()).Times(2).WillRepeatedly(Return(kConnectionKey_));
+  EXPECT_CALL(*app, policy_app_id()).WillOnce(Return(kPolicyAppId_));
+  EXPECT_CALL(*app, hmi_level())
+      .WillOnce(Return(mobile_apis::HMILevel::HMI_NONE));
+
+  EXPECT_CALL(app_mngr_, CheckPolicyPermissions(_, _, _, _, _))
+      .WillOnce(Return(mobile_apis::Result::INVALID_ENUM));
+
+  EXPECT_CALL(app_mngr_, SendMessageToMobile(_, _));
+  EXPECT_FALSE(command->CheckPermissions());
+}
+
+ACTION_P(GetArg3, output) {
+  *output = arg3;
+}
+
+TEST_F(CommandRequestImplTest, CheckAllowedParameters_MsgParamsMap_SUCCESS) {
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::connection_key] = kConnectionKey_;
+  (*msg)[strings::msg_params][kPolicyAppId_] = true;
+
+  CommandPtr command = CreateCommand<UCommandRequestImpl>(msg);
+
+  MockAppPtr app = CreateMockApp();
+  ApplicationSet app_set;
+  app_set.insert(app);
+
+  DataAccessor<ApplicationSet> accessor(app_set, app_set_lock_);
+
+  EXPECT_CALL(app_mngr_, applications()).WillOnce(Return(accessor));
+  EXPECT_CALL(*app, app_id()).WillOnce(Return(kConnectionKey_));
+  EXPECT_CALL(*app, policy_app_id()).WillOnce(Return(kPolicyAppId_));
+  EXPECT_CALL(*app, hmi_level())
+      .WillOnce(Return(mobile_apis::HMILevel::HMI_NONE));
+
+  RPCParams params;
+  EXPECT_CALL(app_mngr_, CheckPolicyPermissions(_, _, _, _, _))
+      .WillOnce(DoAll(GetArg3(&params), Return(kMobResultSuccess_)));
+
+  EXPECT_TRUE(command->CheckPermissions());
+  EXPECT_TRUE(params.end() !=
+              std::find(params.begin(), params.end(), kPolicyAppId_));
+}
+
+TEST_F(CommandRequestImplTest, AddDisallowedParameters_SUCCESS) {
+  const std::string kDisParam =
```

Comment:
@LevchenkoS As for me `DisParam` - is unclear. I would recommend to rename it. 
Dis... - is disabled, disallowed, dismissed or what?